### PR TITLE
raftstore: enable rocksdb's wal

### LINKED ISSUE #851: lose lock info risk
We store lock in separate column family, and we use rocksdb's write_without_wal api to apply raft log. When rocksdb need flush memtable to sstfile, it will choose the column family who's  memtable is bigger than the others, and will only flush this column falimy's memtable to sstfile in this round. This will cause one situation that the default column family's update(data and apply-index info) has flushed to sstfile, but the lock column family's update(lock info) still in memtable, when we shutdown tikv not gracefully, we face the risk of lose lock data.

### PR DESCRIPTION
**Pull request by ngaut**
Fix #851

### CODE REVIEW

**Comment by ngaut** [comment]
PTAL @siddontang @zhangjinpeng1987

File: src/raftstore/store/worker/compact.rs [hunk 1]
```diff
@@ -79,7 +79,7 @@ impl Runner {
         }
         // It is safe to disable WAL here. If crashed, we can still
         // compact the log after restart.
-        box_try!(task.engine.write_without_wal(wb));
+        box_try!(task.engine.write(wb));
```
Comment (siddontang) [inline src/raftstore/store/worker/compact.rs]: I think compact is ok without WAL.

Comment (ngaut) [inline src/raftstore/store/worker/compact.rs]: No, we may left some log in rocksdb after restart.

Comment (siddontang) [inline src/raftstore/store/worker/compact.rs]: no, it will be removed again after restart. because we seek log in [0, truncated index)

Comment (ngaut) [inline src/raftstore/store/worker/compact.rs]: You are right.

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**